tests/unit/test-cmd/test.run.js

```diff
@@ -462,19 +467,42 @@ describe('run', () => {
           assert.ok(addonReload.called);
           assert.equal(addonReload.firstCall.args[0].addonId,
             tempInstallResult.addon.id);
-        });
+        })
+        .then(loopExiter(fakeStdin), loopExiter(fakeStdin));
```

Comment:
The error handler is going to ignore any thrown error which means these tests will always pass which would be bad. You need to make sure it throws the error with something like this:

```diff
diff --git a/tests/unit/test-cmd/test.run.js b/tests/unit/test-cmd/test.run.js
index ef51f95..b91bb2d 100644
--- a/tests/unit/test-cmd/test.run.js
+++ b/tests/unit/test-cmd/test.run.js
@@ -389,6 +389,13 @@ describe('run', () => {
       return () => stdin.emit('keypress', 'c', {name: 'c', ctrl: true});
     }
 
+    function exitLoopOnError(stdin) {
+      return (error) => {
+        loopExiter(stdin);
+        throw error;
+      };
+    }
+
     function prepare() {
       const client = new RemoteFirefox(fakeFirefoxClient());
       const watcher = {
@@ -468,7 +475,7 @@ describe('run', () => {
           assert.equal(addonReload.firstCall.args[0].addonId,
             tempInstallResult.addon.id);
         })
-        .then(loopExiter(fakeStdin), loopExiter(fakeStdin));
+        .then(loopExiter(fakeStdin), exitLoopOnError(fakeStdin));
     });
 
     it('shuts down firefox on user request (CTRL+C in shell console)', () => {
```